packages/sequelize-transparent-cache/src/cache/__test__/util.test.js

```diff
@@ -0,0 +1,136 @@
+const sequelize = require('./sequelize')
+
+const { User, Article, Group } = sequelize.models
+
+beforeAll(() => sequelize.sync())
+afterEach(() => (sequelize.sync({ force: true })))
+
+describe('Recursive include tests', () => {
+  test('Nested include depth == 1', async () => {
+    // get all users of a group
+    const group = await Group.cache().create({
+      id: 1,
+      name: 'Crazy Bloggers'
+    })
+
+    const user1 = await User.cache().create({
+      id: 1,
+      name: 'Bob'
+    })
+
+    const user2 = await User.cache().create({
+      id: 2,
+      name: 'Alice'
+    })
+
+    await group.setGroupUsers([user1, user2])
+    await group.cache().save()
+
+    // From DB
+    const GroupFromDB = await Group.cache('CrazyBloggers').findAll({
+      include: [{
+        model: User,
+        as: 'groupUsers'
+      }],
+      where: { id: 1 }
+    })
+
+    // From cache
+    const cachedGroup = await Group.cache('CrazyBloggers').findAll({
+      include: [{
+        model: User,
+        as: 'groupUsers'
+      }],
+      where: { id: 1 }
+    })
+    // comparing length at depth == 0
+    expect(cachedGroup.length).toEqual(GroupFromDB.length)
+    // comparing dataValues at depth == 0
+    expect(cachedGroup[0].dataValues.id).toEqual(GroupFromDB[0].dataValues.id)
```

Comment:
`expect(cachedGroup[0].get()).toEqual(GroupFromDB[0].get())`

Reply:
Did you mean ```cachedGroup[0].get().id``` and ```GroupFromDB[0].get().id``` and same for the name field. I guess the objects are not exactly identical. This is the output of the test failing if I use ```expect(cachedGroup[0].get()).toEqual(GroupFromDB[0].get())```. Am I missing something?
```


  ● Recursive include tests › Nested include depth == 1

    expect(received).toEqual(expected) // deep equality

    - Expected
    + Received

    @@ -1,25 +1,21 @@
      Object {
        "createdAt": 2020-04-11T12:33:25.101Z,
        "groupUsers": Array [
          Object {
            "UserGroup": Object {
    -         "createdAt": 2020-04-11T12:33:25.124Z,
              "group_id": 1,
    -         "updatedAt": 2020-04-11T12:33:25.124Z,
              "user_id": 1,
            },
            "createdAt": 2020-04-11T12:33:25.111Z,
            "id": 1,
            "name": "Bob",
            "updatedAt": 2020-04-11T12:33:25.111Z,
          },
          Object {
            "UserGroup": Object {
    -         "createdAt": 2020-04-11T12:33:25.124Z,
              "group_id": 1,
    -         "updatedAt": 2020-04-11T12:33:25.124Z,
              "user_id": 2,
            },
            "createdAt": 2020-04-11T12:33:25.114Z,
            "id": 2,
            "name": "Alice",

      48 |     // comparing dataValues at depth == 0
      49 |     console.log(cachedGroup[0].get(), cachedGroup[0].groupUsers[0].get())
    > 50 |     expect(cachedGroup[0].get()).toEqual(GroupFromDB[0].get())
```

Reply:
Take a closer look at objects diff. `toEqual` works perfectly for our need's here, but "createdAt" and "updatedAt" is missing in the cached object. This is the result of issue #53 . Could you please fix it as well?

Reply:
Sure. I'll look into it. However, I think you can merge this PR if no further changes are required. I'll try to resolve issue #53 and do a new PR with the tests updated as you said.

Reply:
This PR already contains a lot of commits and amendments. So adding more work to this would not be the best idea definitely. What do you say?

Reply:
Fair enough, I agree.